### PR TITLE
cli: add -json flag to support job commands

### LINKED ISSUE #1749: CLI should accept JSON job specs
### Nomad version

`v0.4.1`
### Operating system and Environment details

```
Distributor ID: Debian
Description:    Debian GNU/Linux 7.9 (wheezy)
Release:    7.9
Codename:   wheezy
```
### Issue

`invalid key: Job` when submitting JSON jobs 

Trying `nomad run -output <job.nomad>` on an existing HCL job, converting it to JSON, fail with the same error
### Reproduction steps
- Copy the example JSON from https://www.nomadproject.io/docs/jobspec/json.html into  `raw.json`
- Execute `nomad run raw.json`
### Nomad Server logs (if appropriate)

```
-> nomad run raw.json
Error getting job struct: 1 error(s) occurred:

* invalid key: Job
```
### Job file (if appropriate)

The example JSON from the documentation.

Any other JSON job i've tried to submit fail on the same error :) 

```
{
  "Job": {
    "Region": "global",
    "ID": "example",
    "Name": "example",
    "Type": "batch",
    "Priority": 50,
    "AllAtOnce": false,
    "Datacenters": [
      "dc1"
    ],
    "Constraints": [
      {
        "LTarget": "${attr.kernel.name}",
        "RTarget": "linux",
        "Operand": "="
      }
    ],
    "TaskGroups": [
      {
        "Name": "cache",
        "Count": 1,
        "Constraints": null,
        "Tasks": [
          {
            "Name": "redis",
            "Driver": "docker",
            "User": "foo-user",
            "Config": {
              "image": "redis:latest",
              "port_map": [
                {
                  "db": 6379
                }
              ]
            },
            "Constraints": null,
            "Env": {
              "foo": "bar",
              "baz": "pipe"
            },
            "Services": [
              {
                "Name": "cache-redis",
                "Tags": [
                  "global",
                  "cache"
                ],
                "PortLabel": "db",
                "Checks": [
                  {
                    "Id": "",
                    "Name": "alive",
                    "Type": "tcp",
                    "Command": "",
                    "Args": null,
                    "Path": "",
                    "Protocol": "",
                    "Interval": 10000000000,
                    "Timeout": 2000000000
                  }
                ]
              }
            ],
            "Resources": {
              "CPU": 500,
              "MemoryMB": 256,
              "DiskMB": 300,
              "IOPS": 0,
              "Networks": [
                {
                  "ReservedPorts": [
                    {
                      "Label": "rpc",
                      "Value": 25566
                    }
                  ],
                  "DynamicPorts": [
                    {
                      "Label": "db"
                    }
                  ],
                  "MBits": 10
                }
              ]
            },
            "Meta": {
              "foo": "bar",
              "baz": "pipe"
            },
            "KillTimeout": 5000000000,
            "LogConfig": {
              "MaxFiles": 10,
              "MaxFileSizeMB": 10
            },
            "Artifacts": [
              {
                "GetterSource": "http://foo.com/artifact.tar.gz",
                "GetterOptions": {
                  "checksum": "md5:c4aa853ad2215426eb7d70a21922e794"
                },
                "RelativeDest": "local/"
              }
            ]
          }
        ],
        "RestartPolicy": {
          "Interval": 300000000000,
          "Attempts": 10,
          "Delay": 25000000000,
          "Mode": "delay"
        },
        "Meta": {
          "foo": "bar",
          "baz": "pipe"
        }
      }
    ],
    "Update": {
      "Stagger": 10000000000,
      "MaxParallel": 1
    },
    "Periodic": {
      "Enabled": true,
      "Spec": "* * * * *",
      "SpecType": "cron",
      "ProhibitOverlap": true
    },
    "Meta": {
      "foo": "bar",
      "baz": "pipe"
    }
  }
}
```

### LINKED ISSUE #6758: Accept JSON via nomad run
### Nomad version
Nomad v0.9.6

### Operating system and Environment details
Mac + Linux

### Issue
From the Nomad CLI tool I want to be able to output a job spec and then "run" that output.

For example I have a script which stops a running tasks, applies changes to an underlying service, and then restarts it. `nomad job inspect` outputs JSON which I can submit back via the Job HTTP API but then I'm forced to pass Nomad and Vault tokens via CURL and it's a bit messy. I have an environment setup where the `nomad` CLI works and it makes sense to be able to ingest the same format as is output

### Reproduction steps
```shell
$ nomad job inspect my-task > job.json
$ nomad job stop my-task
==> Monitoring evaluation "f5ac2705"
    Evaluation triggered by job "my-task"
    Evaluation status changed: "pending" -> "complete"
==> Evaluation "f5ac2705" finished with status "complete"
$ nomad job run job.json
Error getting job struct: Error parsing job file from job.json: 1 error(s) occurred:

* invalid key: Job
```

I'm open to any alternatives suggested, but basically I want to be able to interact with a job (ie stop + start) from an environment with access to Nomad, but without access to the original jobspec files.

### PR DESCRIPTION
**Pull request by schmichael**
Fixes #6758
Fixes #1749

While the CLI has always supported running JSON jobs, its support has
been via HCLv2's JSON parsing. I have no idea what format it expects the
job to be in, but it's absolutely not in the same format as the API
expects.

So I ignored that and added a new -json flag to explicitly support *API*
style JSON jobspecs.

### CODE REVIEW

**Review by jrasell — APPROVED** [review]
Awesome!!!

**Review by jrasell — CHANGES_REQUESTED** [review]
I missed the test failure on my original review. It looks like the plan command is failing with a redefined flag:
```
panic: job plan flag redefined: hcl1 [recovered]
	panic: job plan flag redefined: hcl1

goroutine 19967 [running]:
testing.tRunner.func1.2({0x22f6e60, 0xc003aae680})
	/usr/local/go/src/testing/testing.go:1209 +0x24e
testing.tRunner.func1()
	/usr/local/go/src/testing/testing.go:1212 +0x218
panic({0x22f6e60, 0xc003aae680})
	/usr/local/go/src/runtime/panic.go:1038 +0x215
flag.(*FlagSet).Var(0xc0042ff920, {0x2ea2d28, 0xc0048eb3b0}, {0x2907e23, 0x4}, {0x0, 0x0})
	/usr/local/go/src/flag/flag.go:879 +0x2f4
flag.(*FlagSet).BoolVar(...)
	/usr/local/go/src/flag/flag.go:632
github.com/hashicorp/nomad/command.(*JobPlanCommand).Run(0xc0048eb300, {0xc004f6f410, 0x3, 0x3})
	/home/circleci/go/src/github.com/hashicorp/nomad/command/job_plan.go:142 +0x237
github.com/hashicorp/nomad/command.TestPlanCommand_Fails(0xc00429b6c0)
	/home/circleci/go/src/github.com/hashicorp/nomad/command/job_plan_test.go:33 +0x198
testing.tRunner(0xc00429b6c0, 0x2a227d8)
	/usr/local/go/src/testing/testing.go:1259 +0x102
created by testing.(*T).Run
	/usr/local/go/src/testing/testing.go:1306 +0x35a
	``

File: .changelog/12591.txt [hunk 1]
```diff
@@ -0,0 +1,2 @@
+```release-note:improvement
+cli: Added -json flag to `nomad job {run,plan,validate}` to support parsing JSON formatted jobs
```
Comment (lgfa29) [inline .changelog/12591.txt:2]: Missing the closing backticks

**Review by lgfa29 — APPROVED** [review]
```console
$ nomad job inspect example | jq '.Job.TaskGroups[0].Count = 2' | nomad job run -json -
```

🎉 

Just needs a quick fix in the CHANGELOG file.

File: .changelog/12591.txt (hunk 1 above)
Comment (schmichael) [reply .changelog/12591.txt:2]: :facepalm:

**Review by jrasell — APPROVED** [review]

**Pull request merged**